### PR TITLE
Fixes #15052: Document issue with openssl versions during rudder upgrade

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/15052

### CODE REVIEW

File: release-data/changelogs/5.0/main.adoc [hunk 1]
```diff
@@ -128,9 +128,26 @@ SLES 10-11, Ubuntu 10.04 LTS-12.04 LTS-13.04-15.10, Windows Server 2008R2-2016,
...
-=== Communication between agents built with 1.1.0 vs. 1.0.x openssl versions
+=== Communication between Rudder server 5.0 and agents 4.3 and older
+
+We have found a communication problem between some versions of rudder
+agent and server due to a bug in OpenSSL library.
+
+The problem occurs when a client using OpenSSL prior to 1.0.2 tries to
+communicate with a server using OpenSSL 1.1.0 or later.
+
+This means that a Rudder agent 4.3 or older on a debian 8, ubuntu 14,
```
Comment (amousset) [inline release-data/changelogs/5.0/main.adoc]: Debian
Ubuntu 14.04
RHEL6
RHEL7
SLES

File: release-data/changelogs/5.0/main.adoc [hunk 2]
```diff
@@ -128,9 +128,26 @@ SLES 10-11, Ubuntu 10.04 LTS-12.04 LTS-13.04-15.10, Windows Server 2008R2-2016,
...
+rhel 6, rhel 7.3 or older, or sles 12SP3 or older cannot communicate
+with Rudder server 5.0 or later.
+
+To work around this problem, you can upgrade the agent in 5.0 first
+since they are compatible with a 4.3 server.
```
Comment (fanf) [inline release-data/changelogs/5.0/main.adoc]: This part is false. 
Tested with:
- server: centos 6.5, rudder 4.3.10 (system open ssl: 1.0.1e-fips)
- agent: centos 7.6, rudder 5.0.15-nightly, (embedded openssl 1.1.1 and 1.1.0 (???), system openssl 1.0.2k-fips)

For the embeded version, not sure how to get it. ldd links to our own lib (/opt/rudder/bin/libssl.so.1.1), but strings say:
```
OPENSSL_1_1_0
OPENSSL_1_1_0d
OPENSSL_1_1_1
OPENSSL_1_1_1a
OPENSSL_1_1_0i
OPENSSL_1_1_0f
```

Comment (amousset) [reply release-data/changelogs/5.0/main.adoc]: `/opt/rudder/bin/openssl version` gives the embedded version number

Comment (fanf) [reply release-data/changelogs/5.0/main.adoc]: Thanks, it's ` 1.1.1b`

**Comment by fanf** [comment]
I propose to change that one since it's now in the upgrade section of the doc and just reference the doc cc @peckpeck @amousset

**Comment by fanf** [comment]
Closing this one as the problem is not present in any rudder supported version anymore - we have better things to do than loose time on it. 

Still, it's very sad that we were not able to make that documentation straight when the need was present for our users.

**Pull request closed without merging**